src/main/java/org/broadinstitute/dsm/jobs/TestBostonUPSTrackingJob.java

```diff
@@ -24,124 +21,186 @@
 import org.slf4j.LoggerFactory;
 import spark.utils.StringUtils;
 
-import java.io.IOException;
 import java.sql.Connection;
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
-import java.sql.SQLException;
+import java.sql.Statement;
 import java.time.Instant;
-import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Base64;
 import java.util.List;
 
 
-public class TestBostonUPSTrackingJob implements BackgroundFunction<PubsubMessage> {
+public class TestBostonUPSTrackingJob implements BackgroundFunction<UPSKit[]> {
 
     private String STUDY_MANAGER_SCHEMA = System.getenv("STUDY_MANAGER_SCHEMA") + ".";
     private String STUDY_SERVER_SCHEMA = System.getenv("STUDY_SERVER_SCHEMA") + ".";
 
     private final Logger logger = LoggerFactory.getLogger(TestBostonUPSTrackingJob.class);
 
-    private String SQL_AVOID_DELIVERED = "and (tracking_to_id is not null or tracking_return_id is not null ) and (kit_shipping_history is null or kit_shipping_history not like \"" + UPSStatus.DELIVERED_TYPE + " %\" or kit_return_history is null or kit_return_history not like \"" + UPSStatus.DELIVERED_TYPE + " %\")" +
-            " order by kit.dsm_kit_request_id ASC";
-
-
     String DELIVERED = "DELIVERED";
     String RECEIVED = "RECEIVED";
 
     private String SELECT_BY_EXTERNAL_ORDER_NUMBER = "and request.external_order_number = ?";
     private Covid19OrderRegistrar orderRegistrar;
+    String endpoint;
+    String username;
+    String password;
+    String accessKey;
+    Connection conn = null;
 
     @Override
-    public void accept(PubsubMessage message, Context context) throws Exception {
-        if (message.data == null) {
+    public void accept(UPSKit[] kitsToLookFor, Context context) throws Exception {
+        if (kitsToLookFor == null || kitsToLookFor.length == 0) {
             logger.info("No message provided");
             return;
         }
-
-
-        logger.info(message.data + " *");
         Config cfg = CFUtil.loadConfig();
-
         String dbUrl = cfg.getString("dsmDBUrl");
-
-        PoolingDataSource<PoolableConnection> dataSource = CFUtil.createDataSource(20, dbUrl);
-
-
+        endpoint = cfg.getString("ups.url");
+        username = cfg.getString("ups.username");
+        password = cfg.getString("ups.password");
+        accessKey = cfg.getString("ups.accesskey");
+        PoolingDataSource<PoolableConnection> dataSource = CFUtil.createDataSource(1, dbUrl);
         logger.info("Starting the UPS lookup job");
 
+        try {
+            conn = dataSource.getConnection();
```

Comment:
If you put `conn` into a try-with-resources, it'll autoclose for you and you can remove the `finally` on L85.